### DISCUSSION TITLE
How to set extension-based colors prioritized over the executable-based color?

### DISCUSSION THREAD

**Discussion by ted1277** [gokcehan/lf#1875] [Q&A] [answered]
I've encountered an issue with `lf` where all files are displayed in white color (`EXEC 0` is used). When I use `ls`, the file color is dependent on its extension (whether it's executable or not), which is the behavior I want in `lf` as well. However, `lf` doesn't seem to respect this, and all files are rendered in white.

I tried setting `ex 0` in `~/.config/lf/colors`, but that didn't seem to help. Is there any way to prioritize file extension-based color settings over executable-based color settings in `lf`?

Any help would be greatly appreciated!

**Comment by CatsDeservePets** [comment]
Hey @ted1277!

Is your problem that you can't change file colours at all? Or is your problem that files marked as executable always use the executable styling over the extension based styling?

If it's the latter, there is an open issue about this: https://github.com/gokcehan/lf/issues/992

**Reply by ted1277** [reply]
It's the latter.

**Reply by CatsDeservePets** [reply, marked as answer]
Hey again!
Like I said, there is currently no agreement on this in https://github.com/gokcehan/lf/issues/992. I totally feel you, this is something that bothers me personally as well.
If you are willing to build `lf` by yourself (which is really easy and fast, thanks to `Go`), you can just update the `icons.go` file like I did here:
https://github.com/CatsDeservePets/lf/commit/61734fd495da52fce495cc70773e7a7bc83f997a

**Reply by ted1277** [reply]
Thanks for sharing your solution. For now, building from source isn't an option for me. Out of curiosity: is the change you made also working for colors, or is it just for icons?

**Reply by CatsDeservePets** [reply]
Oh, good question! You were explicitly asking about colours, not icons.

My previous change only focussed on the `icon`. I created another quick commit to also prioritise the extension colour over the executable one:
https://github.com/CatsDeservePets/lf/commit/652168a8e1862493cb5c564861a60c7bd4bc3ae5

If building from source is not an option for you (which I totally get), I'd suggest commenting on the open issue https://github.com/gokcehan/lf/issues/992. The more people complain about this, the more likely it is that something will be done about it.

**Reply by joelim-work** [reply]
I [posted](https://github.com/gokcehan/lf/issues/992#issuecomment-1371364445) about it in the issue. The main problem seems to be deciding on a design that has the right balance between simplicity and flexibility.
